common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -327,11 +329,10 @@ public UTF8String substringSQL(int pos, int length) {
   /**
    * Returns whether this contains `substring` or not.
    */
-  public boolean contains(final UTF8String substring) {
+  public boolean containsBinary(final UTF8String substring) {
```

Comment:
similar to `equals`/`compareTo`, let's not rename existing functions for backward compatibility, just add comments to say it's the binary version.